tests/integration/s2n_test_common.py

```diff
@@ -0,0 +1,218 @@
+##
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+
+"""
+Common functions to run s2n integration tests.
+"""
+
+import sys
+import subprocess
+import multiprocessing
+import select
+from enum import Enum
+from multiprocessing.pool import ThreadPool
+from s2n_test_constants import *
+from time import sleep
+
+CERT = TEST_ECDSA_CERT
+KEY = TEST_ECDSA_KEY
+
+ALL_VERSIONS = [S2N_SSLv3, S2N_TLS10, S2N_TLS11, S2N_TLS12, S2N_TLS13]
+
+class Cipher():
+    def __init__(self, name, min_version):
+        self.name = name
+        self.min_version = min_version
+
+    def __str__(self):
+        return self.name
+
+ALL_CIPHERS = [
+    Cipher("TLS_AES_256_GCM_SHA384", S2N_TLS13),
+    Cipher("TLS_CHACHA20_POLY1305_SHA256", S2N_TLS13),
+    Cipher("TLS_AES_128_GCM_SHA256", S2N_TLS13)
+]
+
+class Mode(Enum):
+    client = 0
+    server = 1
+
+    def __str__(self):
+        return "Client" if self is Mode.client else "Server"
+
+    def other(self):
+        return Mode.server if self is Mode.client else Mode.client
+
+ALL_MODES = [e.value for e in Mode]
+
+class Status:
+    def __init__(self, name, color):
+        self.name = name
+        self.color = color
+
+    def __str__(self):
+        return with_color(self.name, self.color)
+
+PASSED = Status("PASSED", 32 )
+FAILED = Status("FAILED", 31 )
+
+class Result:
+    def __init__(self, error=None):
+        self.error = error
+        self.status = PASSED if error == None else FAILED
+
+    def __str__(self):
+        result = str(self.status)
+        if self.error:
+            result += "\n\t\t%s %s" % (with_color("Error:", FAILED.color), self.error)
+        return result
+
+class Scenario:
+    def __init__(self, s2n_mode, host, port, version=None, cipher=None, s2n_flags=[], peer_flags=[]):
```

Comment:
Could you add a comment to Scenario? Does this represent a particular instance of a client with specific flags making a connection to a server?